data-prepper-plugins/otel-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/otelmetrics/OTelMetricsGrpcService.java

```diff
@@ -81,7 +78,12 @@ private void processRequest(final ExportMetricsServiceRequest request, final Str
             if (buffer.isByteBuffer()) {
                 buffer.writeBytes(request.toByteArray(), null, bufferWriteTimeoutInMillis);
             } else {
-                buffer.write(new Record<>(request), bufferWriteTimeoutInMillis);
+                Collection<Record<? extends Metric>> metrics;
+                AtomicInteger droppedCounter = new AtomicInteger(0);
+
+                OTelProtoCodec.OTelProtoDecoder oTelProtoDecoder = new OTelProtoCodec.OTelProtoDecoder();
```

Comment:
It looks like the decoder is passed through the constructor in the other sources. We could do the same in this class to avoid instantiating a new one for each request: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceGrpcService.java#L57